src/crs.jl

```diff
@@ -120,12 +120,17 @@ mactype(C::Type{<:CRS}) = numtype(lentype(C))
 """
     isapprox(coords₁, coords₂; kwargs...)
 
-Converts `coords₁` and `coords₂` to `Cartesian` coordinates and compare the coordinate
-values with the `isapprox` method of vectors. The conversion to `Cartesian` coordinates
-takes care of possibly different `Datum`.
+Checks whether or not `coords₁` and `coords₂` are approximately equal,
+i.e., their coordinate values are the same up to floating point precision.
+
+In the case of different datums, converts the datum of `coords₂` to the
+same datum of `coords₁` before comparison.
 """
-Base.isapprox(coords₁::CRS, coords₂::CRS; kwargs...) =
-  isapprox(convert(Cartesian, coords₁), convert(Cartesian, coords₂); kwargs...)
+Base.isapprox(coords₁::CRS{Datum₁}, coords₂::CRS{Datum₂}; kwargs...) where {Datum₁,Datum₂} =
+  isapprox(coords₁, convert(constructor(coords₁), coords₂); kwargs...)
+
+Base.isapprox(coords₁::CRS{Datum}, coords₂::CRS{Datum}; kwargs...) where {Datum} =
```

Comment:
This definition is a bit permissive. It allows comparisons like these:
```
julia> c1 = Mercator(100, 100)
Mercator{WGS84Latest} coordinates
├─ x: 100.0 m
└─ y: 100.0 m

julia> c2 = Robinson(100, 100)
Robinson{WGS84Latest} coordinates
├─ x: 100.0 m
└─ y: 100.0 m

julia> isapprox(c1, c2)
true

julia> c3 = LatLon(30, 60)
GeodeticLatLon{WGS84Latest} coordinates
├─ lat: 30.0°
└─ lon: 60.0°

julia> c4 = AuthalicLatLon(30, 60)
AuthalicLatLon{WGS84Latest} coordinates
├─ lat: 30.0°
└─ lon: 60.0°

julia> isapprox(c3, c4)
true
```
@juliohm, is this behavior intentional in this PR?